airflow/contrib/example_dags/example_gcp_compute_igm.py

```diff
@@ -0,0 +1,137 @@
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+Example Airflow DAG that Uses IGM-type compute operations:
+* copy of Instance Template
+* update of the group manager's templates
+
+This DAG relies on the following Airflow variables
+https://airflow.apache.org/concepts.html#variables
+* PROJECT_ID - Google Cloud Platform project where the Compute Engine instance exists.
+* ZONE - Zone where the Compute Engine instance exists.
+* TEMPLATE_NAME - Name of the template to copy.
+* NEW_TEMPLATE_NAME - name of the new template.
+* NEW_DESCRIPTION - description added to the template
+* SOURCE_TEMPLATE_URL - url of the template to replace
+* DESTINATION_TEMPLATE_URL - url of the new template to replace in the IGM
+* INSTANCE_GROUP_MANAGER_NAME - name of the instance group manager
+"""
+
+import datetime
+
+import airflow
+from airflow import models
+from airflow.contrib.operators.gcp_compute_operator import \
+    GceInstanceTemplateCopyOperator, GceInstanceGroupManagerUpdateTemplateOperator
+
+# [START howto_operator_compute_igm_common_args]
+PROJECT_ID = models.Variable.get('PROJECT_ID', 'polidea-airflow')
+ZONE = models.Variable.get('ZONE', 'europe-west1-b')
+
+default_args = {
+    'start_date': airflow.utils.dates.days_ago(1)
+}
+# [END howto_operator_compute_igm_common_args]
+
+# [START howto_operator_compute_template_copy_args]
+TEMPLATE_NAME = models.Variable.get('TEMPLATE_NAME', 'instance-template-test')
+NEW_TEMPLATE_NAME = models.Variable.get('NEW_TEMPLATE_NAME',
+                                        'instance-template-test-new')
+NEW_DESCRIPTION = models.Variable.get('NEW_DESCRIPTION', 'Test new description')
+GCE_INSTANCE_TEMPLATE_BODY_UPDATE = {
+    "name": NEW_TEMPLATE_NAME,
+    "description": NEW_DESCRIPTION,
+    "properties": {
+        "machineType": "n1-standard-2"
+    }
+}
+# [END howto_operator_compute_template_copy_args]
+
+# [START howto_operator_compute_igm_update_template_args]
+INSTANCE_GROUP_MANAGER_NAME = models.Variable.get('INSTANCE_GROUP_MANAGER_NAME',
+                                                  'instance-group-test')
+
+SOURCE_TEMPLATE_URL = models.Variable.get(
+    'SOURCE_TEMPLATE_URL',
+    "https://www.googleapis.com/compute/beta/projects/"
+    "polidea-airflow/global/instanceTemplates/instance-template-test")
```

Comment:
We probably shouldn't check in your project id

Reply:
Yep. We have implemented a nice way to pass the variables so we might get some abstract defaults (I will send a separate email about it)